memtagabielf64/memtagabielf64.rst

```diff
@@ -0,0 +1,773 @@
+  tags may be "mostly random", e.g. for global variable or heap protection, it
+  may be useful to ensure that adjacent memory regions have different tags to
+  reliably detect linear buffer overflow. In addition, stack tagging
+  instrumentation may require some other non-fully-random optimisations, like
+  using a random base tag per-frame, but intra-frame objects have sequential
+  tags.
+
+General Restrictions
+--------------------
+
+- Global variables may only be tagged in dynamic executables, as assignment of
+  allocation tags happens during load time, and materialization of logical tags
+  happens during relocation processing.
+
+- Global variables may only be tagged on platforms that use RELA relocation
+  types. For more information, see the changed ``R_AARCH64_RELATIVE`` relocation
+  semantics description in this document.
+
+.. raw:: pdf
+
+   PageBreak
+
+Dynamic Section
+===============
+
+MemtagABI adds the following processor-specific dynamic array tags:
+
+.. table:: MemtagABI specific dynamic array tags
+
+  +---------------------------------+------------+--------+-------------------+-------------------+
+  | Name                            | Value      | d\_un  | Executable        | Shared Object     |
+  +=================================+============+========+===================+===================+
+  | ``DT_AARCH64_MEMTAG_MODE``      | 0x70000009 | d\_val | Platform specific | Platform Specific |
+  +---------------------------------+------------+--------+-------------------+-------------------+
+  | ``DT_AARCH64_MEMTAG_HEAP``      | 0x7000000b | d\_val | Platform specific | Platform Specific |
+  +---------------------------------+------------+--------+-------------------+-------------------+
+  | ``DT_AARCH64_MEMTAG_STACK``     | 0x7000000c | d\_ptr | Platform specific | Platform Specific |
+  +---------------------------------+------------+--------+-------------------+-------------------+
+  | ``DT_AARCH64_MEMTAG_GLOBALS``   | 0x7000000d | d\_val | Platform specific | Platform Specific |
+  +---------------------------------+------------+--------+-------------------+-------------------+
+  | ``DT_AARCH64_MEMTAG_GLOBALSSZ`` | 0x7000000f | d\_val | Platform specific | Platform Specific |
+  +---------------------------------+------------+--------+-------------------+-------------------+
+
+``DT_AARCH64_MEMTAG_MODE`` indicates the initial MTE mode that should be set. It
+has two possible values: ``0``, indicating that the desired MTE mode is
+Synchronous, and ``1``, indicating that the desired mode is Asynchonous. This
+entry is only valid on the main executable, usage in dynamically loaded objects
+is ignored.
+
+The presence of the ``DT_AARCH64_MEMTAG_HEAP`` dynamic array entry indicates
+that heap allocations should be protected with memory tagging. Implementation of
+this logic is left to the heap allocator. This entry is only valid on the main
+executable, usage in dynamically loaded objects is ignored.
```

Comment:
"only valid on the main executable" makes ", usage in dynamically loaded objects is ignored" unneeded. Ditto below.

Reply:
Sure, removed.